### PR TITLE
Fixed OpCode::Call to support recursion and added test for it.

### PR DESCRIPTION
**Pull request by nmdis1999**
1. In `vm.rs`:
       a) `OpCode::Call` now uses new local vector to push back result into stack. This is to avoid pushing back instructions on 
             existing locals.
      b)   Added file `tests/files/fibonacci.ukiyo` to test recursion works.

### CODE REVIEW

**Comment by ptersilie** [comment]
Let's get https://github.com/softdevteam/ukiyo_lang/pull/7 in first, and then we can rebase this before I review it.

**Comment by nmdis1999** [comment]
Permision to force push the rebased changes? @ptersilie

**Comment by ptersilie** [comment]
Please do. You can also do any squashes you want to do and which you will make reviewing this PR easier.

**Comment by nmdis1999** [comment]
Ready for review!

File: tests/files/fibonacci.ukiyo [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Run-time:
+//   stdout:
+//      0
+//      1
+//       1
```
Comment (ptersilie) [inline tests/files/fibonacci.ukiyo:5]: I don't remember: does lang_tester not care about misaligned inlining?

File: tests/files/fibonacci.ukiyo [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+}
+func call() {
+    let x = 0;
+    while (x < 10) {
+        print(fib(x));
+        let x = x + 1;
+    }
+}
+
+call();
```
Comment (ptersilie) [inline tests/files/fibonacci.ukiyo:28]: I've noticed that in some files (not all) there's no newline at the end of the file. Might wanna have a look at your editor settings to make sure there's always one.

File: tests/files/while_in_func.ukiyo [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // Run-time:
 //   stdout:
 //     printing value of x
-//     4
+//     3
```
Comment (ptersilie) [inline tests/files/while_in_func.ukiyo:4]: Why was this changed from `4` to `3`? Was the test incorrect before? If so, any idea how bors missed that in the last PR?

**Comment by ptersilie** [comment]
Only some minor comments. Otherwise looks good.

File: tests/files/while_in_func.ukiyo (hunk 3 above)
Comment (nmdis1999) [reply tests/files/while_in_func.ukiyo:4]: that was a mistake, and no bors didn't miss the case in last PR. 
My understanding is that before this PR we extended the arguments passed in function to the local vector in the `vm`.  So any change to `x` was changed trhoughout the program. However in this PR we push it into new locals (which  I think can only be accessed by the function). So essentially I am passing a copy of `x` and not changing the `x`.
I don't know if we should have concept of mutability like that of rust now or not. So need to discuss with you before going forward.

File: tests/files/fibonacci.ukiyo (hunk 1 above)
Comment (nmdis1999) [reply tests/files/fibonacci.ukiyo:5]: It didn't give me any warning or error.

Comment (ltratt) [reply tests/files/fibonacci.ukiyo:5]: It ignores spaces once it's determined the "parent" indentation. But unless the second `1` is indented this way in the output, I'd suggest aligning everythnig.

File: tests/files/while_in_func.ukiyo (hunk 3 above)
Comment (ptersilie) [reply tests/files/while_in_func.ukiyo:4]: Looking at the test I agree this should be `3`, so just change it back. Let's not worry about mutability for now.

File: tests/files/fibonacci.ukiyo (hunk 1 above)
Comment (ptersilie) [reply tests/files/fibonacci.ukiyo:5]: Ok, let's just fix the indentation, if only for aesthetic reasons.

File: tests/files/fibonacci.ukiyo (hunk 2 above)
Comment (nmdis1999) [reply tests/files/fibonacci.ukiyo]: I have added the settings in my editor now. Should be okay going forward.

**Comment by nmdis1999** [comment]
Addressed all comments. Ready for review :)

**Comment by ptersilie** [comment]
Please squash.

**Comment by nmdis1999** [comment]
Done 👍

**Comment by ptersilie** [comment]
> File indentation and removed junk file simplecall copy.

I don't think you need to mention this in the commit message.

**Comment by nmdis1999** [comment]
Edited commit message.
